library/kani/kani_lib.c

```diff
@@ -2,11 +2,82 @@
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
 #include <assert.h>
+#include <stdbool.h>
 #include <stddef.h>
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
 
+// `assert` then `assume`
+#define __KANI_assert(cond, msg)     \
+    do {                             \
+        bool temp = (cond);          \
+        __CPROVER_assert(temp, msg); \
+        __CPROVER_assume(temp);      \
+    } while (0)
+
+// Check that the input is either a power of 2, or 0. Algorithm from Hackers Delight.
+bool __KANI_is_power_of_two(size_t i) { return (i & (i - 1)) == 0; }
```

Comment:
The minimum alignment is 1, so this should check that is `(i != 0)` as well.

Reply:
done